contrib/src/main/java/com/datatorrent/contrib/kafka/POJOKafkaOutputOperator.java

```diff
@@ -0,0 +1,272 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.contrib.kafka;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+
+import org.apache.commons.lang3.ClassUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import com.datatorrent.api.AutoMetric;
+import com.datatorrent.api.Context;
+import com.datatorrent.api.DefaultInputPort;
+import com.datatorrent.lib.util.PojoUtils;
+
+import kafka.producer.KeyedMessage;
+import kafka.producer.ProducerConfig;
+
+/**
+ * POJOKafkaOutputOperator extends from AbstractKafkaOutputOperator receives the POJO
+ * from upstream and converts to Kafka Messages and writes to kafka topic.&nbsp;
+ * <p>
+ * <br>
+ * Ports:<br>
+ * <b>Input</b>: Have only one input port<br>
+ * <b>Output</b>: No Output Port <br>
+ * <br>
+ * Properties:<br>
+ * <b>brokerList</b>: List of brokers in the form of Host1:Port1,Host2:Port2,Host3:port3,...<br>
+ * <b>keyField</b>: Specifies the field creates distribution of tuples to kafka partition. <br>
+ * <b>isBatchProcessing</b>: Specifies whether to write messages in batch or not. By default,
+ *                           the value is true <br>
+ * <b>batchSize</b>: Specifies the batch size.<br>
+ * <br>
+ * <br>
+ * </p>
+ *
+ * @displayName POJO Kafka Output
+ * @category Messaging
+ * @tags Output operator
+ *
+ */
+public class POJOKafkaOutputOperator extends AbstractKafkaOutputOperator<Object,Object>
+{
+  @AutoMetric
+  private long outputMessagesPerSec;
+  @AutoMetric
+  private long outputBytesPerSec;
```

Comment:
If I remember correctly AutoMetric only works when you have setter/getter method for this. @chandnisingh  am I correct?